### ISSUE TITLE
`RuntimeError` in `CoreferenceResolver` if input doc is one word

### ISSUE THREAD

**Issue by itssimon** [explosion/spaCy#11721]
```
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
Cell In [1], line 4
      1 import spacy
      3 nlp = spacy.load("en_coreference_web_trf")
----> 4 doc = nlp("test")

File ~/.envs/clinex/lib/python3.10/site-packages/spacy/language.py:1031, in Language.__call__(self, text, disable, component_cfg)
   1029     raise ValueError(Errors.E109.format(name=name)) from e
   1030 except Exception as e:
-> 1031     error_handler(name, proc, [doc], e)
   1032 if not isinstance(doc, Doc):
   1033     raise ValueError(Errors.E005.format(name=name, returned_type=type(doc)))

File ~/.envs/clinex/lib/python3.10/site-packages/spacy/util.py:1670, in raise_error(proc_name, proc, docs, e)
   1669 def raise_error(proc_name, proc, docs, e):
-> 1670     raise e

File ~/.envs/clinex/lib/python3.10/site-packages/spacy/language.py:1026, in Language.__call__(self, text, disable, component_cfg)
   1024     error_handler = proc.get_error_handler()
   1025 try:
-> 1026     doc = proc(doc, **component_cfg.get(name, {}))  # type: ignore[call-arg]
   1027 except KeyError as e:
   1028     # This typically happens if a component is not initialized
   1029     raise ValueError(Errors.E109.format(name=name)) from e

File ~/.envs/clinex/lib/python3.10/site-packages/spacy/pipeline/trainable_pipe.pyx:56, in spacy.pipeline.trainable_pipe.TrainablePipe.__call__()

File ~/.envs/clinex/lib/python3.10/site-packages/spacy/util.py:1670, in raise_error(proc_name, proc, docs, e)
   1669 def raise_error(proc_name, proc, docs, e):
-> 1670     raise e

File ~/.envs/clinex/lib/python3.10/site-packages/spacy/pipeline/trainable_pipe.pyx:52, in spacy.pipeline.trainable_pipe.TrainablePipe.__call__()

File ~/.envs/clinex/lib/python3.10/site-packages/spacy_experimental/coref/coref_component.py:148, in CoreferenceResolver.predict(self, docs)
    146 out = []
    147 for doc in docs:
--> 148     scores, idxs = self.model.predict([doc])
    149     # idxs is a list of mentions (start / end idxs)
    150     # each item in scores includes scores and a mapping from scores to mentions
    151     ant_idxs = idxs

File ~/.envs/clinex/lib/python3.10/site-packages/thinc/model.py:315, in Model.predict(self, X)
    311 def predict(self, X: InT) -> OutT:
    312     """Call the model's `forward` function with `is_train=False`, and return
    313     only the output, instead of the `(output, callback)` tuple.
    314     """
--> 315     return self._func(self, X, is_train=False)[0]

File ~/.envs/clinex/lib/python3.10/site-packages/thinc/layers/chain.py:55, in forward(model, X, is_train)
     53 callbacks = []
     54 for layer in model.layers:
---> 55     Y, inc_layer_grad = layer(X, is_train=is_train)
     56     callbacks.append(inc_layer_grad)
     57     X = Y

File ~/.envs/clinex/lib/python3.10/site-packages/thinc/model.py:291, in Model.__call__(self, X, is_train)
    288 def __call__(self, X: InT, is_train: bool) -> Tuple[OutT, Callable]:
    289     """Call the model's `forward` function, returning the output and a
    290     callback to compute the gradients via backpropagation."""
--> 291     return self._func(self, X, is_train=is_train)

File ~/.envs/clinex/lib/python3.10/site-packages/spacy_experimental/coref/coref_model.py:85, in coref_forward(model, X, is_train)
     84 def coref_forward(model: Model, X, is_train: bool):
---> 85     return model.layers[0](X, is_train)

File ~/.envs/clinex/lib/python3.10/site-packages/thinc/model.py:291, in Model.__call__(self, X, is_train)
    288 def __call__(self, X: InT, is_train: bool) -> Tuple[OutT, Callable]:
    289     """Call the model's `forward` function, returning the output and a
    290     callback to compute the gradients via backpropagation."""
--> 291     return self._func(self, X, is_train=is_train)

File ~/.envs/clinex/lib/python3.10/site-packages/thinc/layers/pytorchwrapper.py:143, in forward(model, X, is_train)
    140 convert_outputs = model.attrs["convert_outputs"]
    142 Xtorch, get_dX = convert_inputs(model, X, is_train)
--> 143 Ytorch, torch_backprop = model.shims[0](Xtorch, is_train)
    144 Y, get_dYtorch = convert_outputs(model, (X, Ytorch), is_train)
    146 def backprop(dY: Any) -> Any:

File ~/.envs/clinex/lib/python3.10/site-packages/thinc/shims/pytorch.py:72, in PyTorchShim.__call__(self, inputs, is_train)
     70     return self.begin_update(inputs)
     71 else:
---> 72     return self.predict(inputs), lambda a: ...

File ~/.envs/clinex/lib/python3.10/site-packages/thinc/shims/pytorch.py:90, in PyTorchShim.predict(self, inputs)
     88 with torch.no_grad():
     89     with torch.cuda.amp.autocast(self._mixed_precision):
---> 90         outputs = self._model(*inputs.args, **inputs.kwargs)
     91 self._model.train()
     92 return outputs

File ~/.envs/clinex/lib/python3.10/site-packages/torch/nn/modules/module.py:1190, in Module._call_impl(self, *input, **kwargs)
   1186 # If we don't have any hooks, we want to skip the rest of the logic in
   1187 # this function, and just call forward.
   1188 if not (self._backward_hooks or self._forward_hooks or self._forward_pre_hooks or _global_backward_hooks
   1189         or _global_forward_hooks or _global_forward_pre_hooks):
-> 1190     return forward_call(*input, **kwargs)
   1191 # Do not call functions when jit is used
   1192 full_backward_hooks, non_full_backward_hooks = [], []

File ~/.envs/clinex/lib/python3.10/site-packages/spacy_experimental/coref/pytorch_coref_model.py:74, in CorefClusterer.forward(self, word_features)
     70 words = self.dropout(words)
     71 # Obtain bilinear scores and leave only top-k antecedents for each word
     72 # top_rough_scores: (n_words x rough_k)
     73 # top_indices: (n_words x rough_k)
---> 74 top_rough_scores, top_indices = self.rough_scorer(words)
     75 # Get pairwise features
     76 # (n_words x rough_k x n_pairwise_features)
     77 pairwise = self.pairwise(top_indices)

File ~/.envs/clinex/lib/python3.10/site-packages/torch/nn/modules/module.py:1190, in Module._call_impl(self, *input, **kwargs)
   1186 # If we don't have any hooks, we want to skip the rest of the logic in
   1187 # this function, and just call forward.
   1188 if not (self._backward_hooks or self._forward_hooks or self._forward_pre_hooks or _global_backward_hooks
   1189         or _global_forward_hooks or _global_forward_pre_hooks):
-> 1190     return forward_call(*input, **kwargs)
   1191 # Do not call functions when jit is used
   1192 full_backward_hooks, non_full_backward_hooks = [], []

File ~/.envs/clinex/lib/python3.10/site-packages/spacy_experimental/coref/pytorch_coref_model.py:238, in RoughScorer.forward(self, mentions)
    236 pair_mask = torch.log((pair_mask > 0).to(torch.float))
    237 pair_mask = pair_mask.to(mentions.device)
--> 238 bilinear_scores = self.dropout(self.bilinear(mentions)).mm(mentions.T)
    239 rough_scores = pair_mask + bilinear_scores
    240 top_scores, indices = torch.topk(
    241     rough_scores, k=min(self.k, len(rough_scores)), dim=1, sorted=False
    242 )

RuntimeError: self must be a matrix
```

## How to reproduce the behaviour
```shell
pip install https://github.com/explosion/spacy-experimental/releases/download/v0.6.0/en_coreference_web_trf-3.4.0a0-py3-none-any.whl
```
```python
import spacy

nlp = spacy.load("en_coreference_web_trf")
doc = nlp("test")
```

## Your Environment
* Operating System: Linux
* Python Version Used: 3.10
* spaCy Version Used: 3.4.2 (and spacy-experimental 0.6.0)

**Comment by polm** [comment]
Thanks for the report! I was able to reproduce it, we'll work on fixing it.

**Comment by kadarakos** [comment]
Thank you for the bug report. It makes sense to implement special handling for single token documents as the definition of coreference resolution requires at least two tokens.

**Comment by polm** [comment]
Still checking it, but there's a PR with a fix here: https://github.com/explosion/spacy-experimental/pull/28

**Comment by polm** [comment]
This should be resolved in the recent [0.6.1 release of `spacy-experimental`](https://github.com/explosion/spacy-experimental/releases/tag/v0.6.1).
